mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td

```diff
@@ -25,9 +25,11 @@ include "mlir/Dialect/LLVMIR/LLVMTypes.td"
 def LLVM_PointerGeneric : LLVM_PointerInAddressSpace<0>;
 def LLVM_PointerGlobal : LLVM_PointerInAddressSpace<1>;
 def LLVM_PointerShared : LLVM_PointerInAddressSpace<3>;
+def LLVM_PointerConst : LLVM_PointerInAddressSpace<4>;
 def LLVM_PointerLocal : LLVM_PointerInAddressSpace<5>;
 def LLVM_PointerTensor : LLVM_PointerInAddressSpace<6>;
 def LLVM_PointerSharedCluster : LLVM_PointerInAddressSpace<7>;
+def LLVM_PointerParam : LLVM_PointerInAddressSpace<101>;
```

Comment:
I am not sure what is this :)

Reply:
The address spaces are defined across the llvm, clang, mlir. 

How do others handle param space?

Reply:
Isn't 101 always meant to be internal? I don't remember anywhere we expose this to the user

Reply:
Let's go with an example. Who is lowering to `%param_ptr: !llvm.ptr<101>` today in MLIR? 
```
llvm.func @prefetch_tensormap(%param_ptr: !llvm.ptr<101>) {
  nvvm.prefetch.tensormap %param_ptr: !llvm.ptr<101>
  llvm.return
}
```

alternative idea is to have op specific param space like following
```
llvm.func @prefetch_tensormap(%param_ptr: !llvm.ptr) {
  nvvm.prefetch.tensormap %param_ptr {in_param_space}: !llvm.ptr
  llvm.return
}
```